charts/stardog/files/utils.sh

```diff
@@ -32,7 +32,25 @@ function change_pw {
     PORT=${2}
     NEW_PW=$(cat /etc/stardog-password/adminpw)
     /opt/stardog/bin/stardog-admin --server http://${HOST}:${PORT} user passwd -N ${NEW_PW}
-    RC=$?
-    return ${RC}
+    if [[ $? -eq 0 ]];
+    then
+	    echo "Password successfully changed"
+	    return 0
+    else
+    	curl --fail -u admin:${NEW_PW} http://${HOST}:${PORT}/admin/status
+    	RC=$?
+    	if [[ $RC -eq 0 ]];
+      then
+        echo "Default password was already changed"
+        return 0
+      elif [[ $RC -eq 22 ]]
+      then
+        echo "Authentication failed"
+        return 1
```

Comment:
One more small nitpick here. For curl a return code of 22 is just a 400+ response, so it could be other things, though admittedly not likely a lot of other things. Let's update the message to reflect that to avoid confusion if it does happen. Also we can return $rc here instead of 1.